app/db/db.py

```diff
@@ -0,0 +1,244 @@
+    try:
+        cur = con.cursor()
+        sqlUser = """CREATE TABLE IF NOT EXISTS users(
+        ID SERIAL PRIMARY KEY,
+        FIRSTNAME VARCHAR(50) NOT NULL,
+        LASTNAME VARCHAR(50) NOT NULL,
+        EMAIL VARCHAR(80) NOT NULL,
+        BIRTHDATE VARCHAR(50) NOT NULL,
+        LOCATION VARCHAR(50) NOT NULL,
+        PASSWORD VARCHAR(80) NOT NULL);
+        """
+        sqlRide="""
+        CREATE TABLE IF NOT EXISTS rides(
+        ID SERIAL PRIMARY KEY,
+        START VARCHAR(50) NOT NULL,
+        FINISH VARCHAR(50) NOT NULL,
+        DEPARTURE_DATE VARCHAR(50) NOT NULL,
+        EMAIL VARCHAR(50) NOT NULL,
+        SLOT INT NOT NULL,
+        USER_ID INT REFERENCES USERS(ID)
+        )
+        """
+        cur.execute(sqlUser)
+        cur.execute(sqlRide)
+        con.commit()
+    except psycopg2.Error as e:
+        con.rollback()
+        return({e.pgcode: e.pgerror}, 500)
+
+    firstName, lastName, location = data['firstName'], data['lastName'], data['location']
+    password, email, birthDate = data['password'], data['email'], data['birthDate']
+    newPassword = generate_password_hash(password, method='sha256')
+    try:
+        cur = con.cursor()
+        sql = "INSERT INTO users "\
+        "(firstName, lastName, email, location, birthDate, password) VALUES "\
+        "('{}', '{}', '{}', '{}', '{}', '{}')".format(firstName, lastName, email, location, birthDate, newPassword)
+        cur.execute(sql)
+        con.commit()
+        return({'result': 'account created'},201)
+    except psycopg2.Error as e:
+        con.rollback()
+        return({e.pgcode: e.pgerror}, 500)
+
+
+# check if email exist
+
+
+def emailExist(email):
+    try:
+        cur = con.cursor()
+        sqlUser = """CREATE TABLE IF NOT EXISTS users(
+        ID SERIAL PRIMARY KEY,
+        FIRSTNAME VARCHAR(50) NOT NULL,
+        LASTNAME VARCHAR(50) NOT NULL,
+        EMAIL VARCHAR(80) NOT NULL,
+        BIRTHDATE VARCHAR(50) NOT NULL,
+        LOCATION VARCHAR(50) NOT NULL,
+        PASSWORD VARCHAR(80) NOT NULL);
+        """
+        sqlRide="""
+        CREATE TABLE IF NOT EXISTS rides(
+        ID SERIAL PRIMARY KEY,
+        START VARCHAR(50) NOT NULL,
+        FINISH VARCHAR(50) NOT NULL,
+        DEPARTURE_DATE VARCHAR(50) NOT NULL,
+        EMAIL VARCHAR(50) NOT NULL,
+        SLOT INT NOT NULL,
+        USER_ID INT REFERENCES USERS(ID)
+        )
+        """
+        cur.execute(sqlUser)
+        cur.execute(sqlRide)
+        con.commit()
+    except psycopg2.Error as e:
+        con.rollback()
+        return({e.pgcode: e.pgerror}, 500)
+
+    try:
+        cur = con.cursor()
+        sql = "SELECT * FROM users WHERE EMAIL='{}'".format(email)
+        cur.execute(sql)
+        item = cur.fetchall()
+        if item:
+            return(True)
+        else:
+            return(False)
+    except TypeError:
+        return{'result': 'Invalid data type'}, 404
+
+# check user credentials
+
+
+def loginUser(email, password):
```

Comment:
Why are we creating a table to log in a user?

Reply:
So that in the event user is trying to log in and the tables do not exist the the tables will be created